pgml-extension/src/bindings/transformers/transformers.py

```diff
@@ -500,6 +500,33 @@ def transform(task, args, inputs, stream=False):
     return orjson.dumps(pipe(inputs, **args), default=orjson_default).decode()
 
 
+def create_cross_encoder(transformer):
+    return CrossEncoder(transformer)
+
+
+def rank_using(model, query, documents, kwargs):
+    if isinstance(kwargs, str):
+        kwargs = orjson.loads(kwargs)
+
+    # The score is a numpy float32 before we convert it
+    return [
+        {"score": x.pop("score").item(), **x}
+        for x in model.rank(query, documents, **kwargs)
+    ]
+
+
+def rank(transformer, query, documents, kwargs):
+    kwargs = orjson.loads(kwargs)
+
+    if transformer not in __cache_sentence_transformer_by_name:
+        __cache_sentence_transformer_by_name[transformer] = create_cross_encoder(
+            transformer
```

Comment:
If you pass kwargs through to `create_cross_encoder` we can specify the device, https://www.sbert.net/docs/package_reference/cross_encoder/cross_encoder.html?highlight=crossencoder#sentence_transformers.cross_encoder.CrossEncoder

We should do this for the `SentenceTransformer` constructor, too.

Reply:
We will have to create a separate argument for this, or pop specific arguments from kwargs. If just pass kwargs straight through we will get an unexpected keyword argument error.